File: backend/src/models/jurisdiction.py
"""Models for jurisdiction comparison data."""

from enum import StrEnum

from pydantic import BaseModel


class ServiceFilter(StrEnum):
    custody = "custody"
    exchange = "exchange"
    trading_platform = "trading_platform"
    advice = "advice"
    portfolio_management = "portfolio_management"
    transfer = "transfer"
    all = "all"


class LicenseType(BaseModel):
    type: str
    description: str
    article: str


class CapitalRequirement(BaseModel):
    capital_class: str
    amount: str
    services: str


class JurisdictionComparison(BaseModel):
    name: str
    regulatory_body: str
    framework: str
    license_types: list[LicenseType]
    capital_requirements: list[CapitalRequirement]
    timeline: str
    passporting: str
    aml_requirements: list[str]
    key_obligations: list[str]
    transitional_provisions: str
    recent_developments: str


class JurisdictionListResponse(BaseModel):
    jurisdictions: list[JurisdictionComparison]

File: frontend/src/types/jurisdiction.ts
export interface JurisdictionLicenseType {
  type: string
  description: string
  article: string
}

export interface JurisdictionCapitalRequirement {
  capitalClass: string
  amount: string
  services: string
}

export interface JurisdictionInfo {
  name: string
  regulatoryBody: string
  framework: string
  licenseTypes: JurisdictionLicenseType[]
  capitalRequirements: JurisdictionCapitalRequirement[]
  timeline: string
  passporting: string
  amlRequirements: string[]
  keyObligations: string[]
  transitionalProvisions: string
  recentDevelopments: string
}

export interface JurisdictionsResponse {
  jurisdictions: JurisdictionInfo[]
}
